src/middlewared/middlewared/plugins/reporting/events.py

```diff
@@ -160,6 +161,31 @@ def run(self):
                             }
                             data['interfaces'][iface.name].update(details_dict)
                         last_interface_stats[iface.name] = {**data['interfaces'][iface.name], 'stats_time': stats_time}
+            else:
+                stats_time = time.time()
+                for i in glob.glob('/sys/class/net/*/statistics'):
+                    iface_name = i.replace('/sys/class/net/', '').split('/')[0]
+                    for stat, name in retrieve_stat.items():
+                        with open(f'{i}/{stat}', 'r') as f:
+                            value = int(f.read())
+                        data['interfaces'][iface_name][name] = value
+                        traffic_stats = None
+                        if (
+                            last_interface_stats.get(iface_name) and
+                            name in last_interface_stats[iface_name]
+                        ):
+                            traffic_stats = value - last_interface_stats[iface_name][name]
+                            traffic_stats = int(
+                                traffic_stats / (
+                                    stats_time - last_interface_stats[iface_name]['stats_time']
```

Comment:
Won't it be better to use `monotime` for calculations like this?

Reply:
It would, it just copied the block/logic from the freebsd side though